### ISSUE TITLE
android wrong time

### ISSUE THREAD

**Issue by RMabroukS** [zo0r/react-native-push-notification#1464]
# Question

I have an app that sends notification for prayer times ,
But I receive notifications at random times
for Example : 
    PushNotification.localNotificationSchedule({
        id: "1",
        largeIcon: "ic_launcher", 
        smallIcon: "ic_notification",
        vibrate: true,
        vibration: 300,
        priority: "high",
        visibility: "private", 
        importance: "high", 
        allowWhileIdle: false, 
        ignoreInForeground: false, 
        alertAction: "view", 
        category: "", 
        userInfo: {}, 
        title: title, 
        message: message,
        playSound: true, 
        soundName: "sound.mp3",
        date: new Date().setMinutes(10),  //Fri Jun 05 2020 22:10:00 GMT+0300 (Arabian Standard Time)
    });

Expected to be received at : 22:10:00
But I receive at random times ,Some times :22:12:00 and Some times :22:13:00 and Some times :22:33:00 There are no fixed times
But the same code works on iOS and receive at the same time .
 I don’t know what the problem is

"dependencies": {
    "@react-native-community/async-storage": "^1.7.1",
    "@react-native-community/audio-toolkit": "^2.0.3",
    "@react-native-community/cli-debugger-ui": "3.0.0",
    "@react-native-community/geolocation": "^2.0.2",
    "@react-native-community/masked-view": "^0.1.6",
    "@react-native-community/push-notification-ios": "^1.1.1",
    "@react-native-community/viewpager": "^3.3.0",
    "@react-native-firebase/app": "^6.7.1",
    "@react-native-firebase/messaging": "^6.7.1",
    "@react-navigation/bottom-tabs": "^5.1.0",
    "@react-navigation/drawer": "^5.1.0",
    "@react-navigation/native": "5.0.0",
    "@react-navigation/stack": "5.0.0",
    "apisauce": "^1.1.1",
    "lottie-react-native": "^3.3.2",
    "moment": "^2.24.0",
    "moment-hijri": "^2.1.2",
    "react": "16.11.0",
    "react-native": "0.62.2",
    "react-native-device-heading": "^1.1.0",
    "react-native-device-info": "^5.5.6",
    "react-native-geocoding": "^0.4.0",
    "react-native-geolocation-service": "^4.0.0",
    "react-native-gesture-handler": "^1.6.0",
    "react-native-i18n": "^2.0.15",
    "react-native-linear-gradient": "^2.5.6",
    "react-native-maps": "^0.27.1",
    "react-native-maps-directions": "^1.8.0",
    "react-native-modal": "^11.5.6",
    "react-native-open-map": "^3.0.2",
    "react-native-open-maps": "^0.3.5",
    "react-native-pdf": "^6.1.1",
    "react-native-permissions": "^2.1.2",
    "react-native-progress-circle": "^2.1.0",
    "react-native-push-notification": "^3.5.2",
    "react-native-reanimated": "^1.7.0",
    "react-native-restart": "0.0.15",
    "react-native-safe-area-context": "^0.7.3",
    "react-native-screens": "2.0.0-beta.2",
    "react-native-search-filter": "^0.1.5",
    "react-native-sound": "^0.11.0",
    "react-native-svg": "^11.0.1",
    "react-native-svg-uri": "^1.2.3",
    "react-native-swiper": "^1.6.0-nightly.5",
    "react-native-vector-icons": "^6.6.0",
    "react-navigation": "^4.1.1",
    "react-redux": "^7.1.3",
    "redux": "^4.0.5",
    "redux-logger": "^3.0.6",
    "redux-persist": "^6.0.0",
    "redux-saga": "^1.1.3",
    "rn-fetch-blob": "^0.12.0"
  },

**Comment by HyeonjuPark** [comment]
I had a same problem. Also, sometimes I got the push when I enter the app after the reservation time.

**Comment by RMabroukS** [comment]
@HyeonjuPark  Did you find a solution to this problem?

**Comment by HyeonjuPark** [comment]
@RMabroukS no, but I cleared all scheduled notifications and re-reserved all again, then it worked again. I don't know why.

**Comment by Dallas62** [comment]
Hi,
Maybe tou should use `allowWhileIdle` if the time is important.

**Comment by HyeonjuPark** [comment]
@Dallas62 Is `allowWhileIdle` related to the issue that notification is not came in exact time?

**Comment by RMabroukS** [comment]
@Dallas62 @HyeonjuPark I set it to “true” but there is no effect , the problem still exists!
 But I think the problem is related to the android V 10 ، They turned off notifications when the screen is locked 🙁

**Comment by HyeonjuPark** [comment]
@RMabroukS @Dallas62 I think I found the cause why the notification is not come in right time. There is an option to optimize device memory in Samsung Galaxy or some devices (relatively recent devices). If this option is activated, device controls background service (even scheduled local notification) and don't allow to show notifications.
check this link
https://dontkillmyapp.com/samsung

**Comment by kdpadhiyar** [comment]
> @Dallas62 @HyeonjuPark I set it to “true” but there is no effect , the problem still exists!
> But I think the problem is related to the android V 10 ، They turned off notifications when the screen is locked 🙁

You can change it by going to app info and do enable it. If notification not appear when phone is locked.Or if your app is running in power saving or battery optimized performance mode than change it to no restriction.
